python/ray/experimental/serve/kv_store_service.py

```diff
@@ -0,0 +1,173 @@
+import json
+from abc import ABC
+
+import ray
+import ray.experimental.internal_kv as ray_kv
+from ray.experimental.serve.utils import logger
+
+
+class NamespacedKVStore(ABC):
+    """Abstract base class for a namespaced key-value store.
+    The idea is that multiple key-value stores can be created while sharing
+    the same database connection. The keys of each instance are namespaced
+    for conflict avoidance.
+
+    Usage:
+
+    >>> store_ns1 = NamespacedKVStore(namespace="ns1")
+    >>> store_ns2 = NamespacedKVStore(namespace="ns2")
+    # Two store can share the same connection like Redis or SQL Table
+    >>> store_ns1.put("same-key", 1)
+    >>> store_ns1.get("same-key")
+    1
+    >>> store_ns2.put("same-key", 2)
+    >>> store_ns2.get("same-key", 2)
+    2
+    """
+
+    def __init__(self, namespace):
+        raise NotImplementedError()
+
+    def get(self, key):
+        """Retrieve the value given key
+
+        Args:
+            key (str)
+        """
+        raise NotImplementedError()
+
+    def put(self, key, value):
+        """Store the correponding entry key -> value.
+
+        Args:
+            key (str)
+            value (object): any serializable object. The serialization method
+                            is determined by the subclass implementation.
+        """
+        raise NotImplementedError()
+
+    def as_dict(self):
+        """Return the entire namespace as python dictionary.
+
+        Returns:
+            data (dict): key value pairs in current namespace
+        """
+        raise NotImplementedError()
+
+
+class InMemoryKVStore(NamespacedKVStore):
```

Comment:
Is this necessary? I feel like the class above is pretty clear, and there's literally an implemented example below.

Reply:
`InMemoryKVStore` can be used for testing

Reply:
is it currently used for testing? if so, comment that it is. if not, remove it.